core/controller/src/main/scala/whisk/core/controller/Meta.scala

```diff
@@ -44,27 +151,35 @@ trait WhiskMetaApi extends Directives with PostActionActivation {
     /** Store for identities. */
     protected val authStore: AuthStore
 
-    /** The route prefix e.g., /experimental/package-name. */
-    protected val routePrefix = pathPrefix("experimental")
+    /** The route prefix e.g., /experimental. */
+    protected val routePath = "experimental"
+
+    /** The prefix for anonymous invokes e.g., /experimental/web. */
+    protected val anonymousInvokePath = "web"
 
     /** The name and apikey of the system namespace. */
     protected val systemId = "whisk.system"
     protected lazy val systemNamespace = EntityPath(systemId)
     protected lazy val systemKey = WhiskAuth.get(authStore, Subject(systemId), false)(TransactionId.controller)
 
-    /** Reserved parameters that requests may no defined. */
-    protected lazy val reservedProperties = Set("__ow_meta_verb", "__ow_meta_path", "__ow_meta_namespace")
+    private val routePrefix = pathPrefix(routePath)
+    private val anonymousInvokePrefix = pathPrefix(anonymousInvokePath)
 
     /** Allowed verbs. */
     private lazy val allowedOperations = get | delete | post
 
-    /** Extracts the HTTP method, query params and unmatched (remaining) path. */
+    private lazy val validNameSegment = pathPrefix(EntityName.REGEX.r)
+    private lazy val packagePrefix = pathPrefix("default".r | EntityName.REGEX.r)
+    private lazy val extensionSplitter = EntityName.REGEX.dropRight(2).concat("""\.([a-zA-Z0-9.+-]+)\z""").r
```

Comment:
Rather not mess with it because I'll have to think about the implications of that (i.e., where are all the places we use that regex match). I think being explicit does not hurt (i.e., specifying the anchors and make it clear this is a full match). I should add a comment to explain the drop right though in that case.